### PR TITLE
fix: Make manifest Swift 6 Language Mode compatible

### PR DESCRIPTION
**Pull request by jbelkins**
## Description of changes
Modifies the AWS SDK package manifest to not raise Swift 6 concurrency warning/errors.
- Generated code is now at the top of the manifest, right under the headers & imports.
- Generated code is now all immutable, `Sendable` globally-spaced variables.  There are no more mutable variables in the manifest.
- The package manifest is constructed by calling functions that transform the generated code to create & return the manifest components.  Since the generated code components are all immutable & `Sendable`, there is no concurrency issue with calling these functions.

The `Package.swift` in this PR is the new, re-rendered version.

Also:
- Eliminated the ability to use `main` branch of `smithy-swift` which is unused.
- Eliminated the manifest generator's `--exclude-aws-services` CLI option, which no longer has utility since integration tests & protocol tests were moved to their own Swift packages.

## New/existing dependencies impact assessment, if applicable
No new dependencies were added to this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/ci_steps/prepare_protocol_and_unit_tests.sh [hunk 1]
```diff
@@ -2,8 +2,12 @@
 
 set -e
 
+# Get rid of the generated services
+rm -rf Sources/Services/*
+
+# Regenerate the SDK manifest, with only runtime included
 cd AWSSDKSwiftCLI
-swift run AWSSDKSwiftCLI generate-package-manifest --exclude-aws-services ..
+swift run AWSSDKSwiftCLI generate-package-manifest ..
```
Comment (jbelkins) [inline scripts/ci_steps/prepare_protocol_and_unit_tests.sh:10]: Simply deleting the services & regenerating the manifest has the same effect as the now-deleted CLI option.

File: AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Prefix.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// swift-tools-version:5.9
```
Comment (jbelkins) [inline AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Prefix.txt:1]: Generated code now goes above the static content, so this separate "prefix" file is needed to provide the copyright notice, imports, etc.

File: AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.txt [hunk 3]
```diff
@@ -1,19 +1,6 @@
-// swift-tools-version:5.9
+// MARK: - Static Content
```
Comment (jbelkins) [inline AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.txt:1]: This static file gets rendered into the manifest after the dynamic content.

File: AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.txt [hunk 4]
```diff
@@ -56,15 +43,50 @@ let package = Package(
...
+        runtimeProducts +
+        serviceTargets.map(productForService(_:)),
+    dependencies:
+        [clientRuntimeDependency, crtDependency, doccDependencyOrNil].compactMap { $0 },
+    targets:
+        runtimeTargets +
+        runtimeTestTargets +
+        serviceTargets.map(target(_:)) +
+        serviceTargets.map(unitTestTarget(_:))
+)
```
Comment (jbelkins) [inline AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.txt:56]: The `Package` instance for this manifest is above.  Note that its params are either a literal or some combination of function or read-only var results.

**Review by sichanyoo — APPROVED** [review]

**Pull request merged**